### ISSUE TITLE
Column resolution fails in complex join conditions 

### ISSUE THREAD

**Issue by bziobrowski** [questdb/questdb#2329]
### Describe the bug

If query contains a with clause qdb sometimes fails to resolve columns in complex expressions . 
Column is properly resolved if it's accessed directly , not through WITH 'view' .

### To reproduce

```sql
CREATE TABLE 'prices' (ts TIMESTAMP, price DOUBLE) timestamp (ts);

with qwe as (select * from prices),
  select p1.ts, p2.price
  from qwe p1
  left join qwe p2 on p2.ts - p1.ts = 0 or 
                        p2.ts - p1.ts + 3600000000 = 0 or 
                        p2.ts - p1.ts + 2*3600000000 = 0 or 
                        p2.ts - p1.ts + 3*3600000000 = 0 or 
                        p2.ts - p1.ts + 4*3600000000 = 0 or 
                        p2.ts - p1.ts + 5*3600000000 = 0 or 
                        p2.ts - p1.ts + 6*3600000000 = 0 or 
                        p2.ts - p1.ts + 7*3600000000 = 0 or 
                        p2.ts - p1.ts + 8*3600000000 = 0 or 
                        p2.ts - p1.ts + 9*3600000000 = 0 
  where p1.ts between '1970-01-01T01:00:00.000000Z' and '1971-01-01T01:00:00.000000Z' 
  and   p2.ts between '1970-01-01T01:00:00.000000Z' and '1971-01-01T01:00:00.000000Z' 
  limit 10000
```

### Environment

```markdown
- **QuestDB version**: 6.4.3
- **OS**: Windows
- **Browser**: Chrome
```

**Comment by tranquilo12** [comment]
I've also encountered another similar issue.

I've got a table full of aggregates called 'aggs', and I've got a table full of urls called 'urls' that are used to pull down the aggregates from a data provider.

I use a small program to check which urls from 'urls' have not been used to populate 'aggs' yet, and then use them to fill up 'aggs'.
But I need a query to determine which urls are "done", by checking 'aggs', as a sanity check.So I just update the table 'urls', based on what's in aggregates using the following query, or so I thought.

The below query does not work, it throws a Invalid column: q.timestamp  error ONLY when I include both the conditions.

```sql
WITH q AS (
  SELECT ticker, timestamp
  FROM aggs 
  LATEST ON timestamp 
  PARTITION BY ticker
)
UPDATE urls u
SET done = true
FROM q
WHERE q.timestamp <= u.end
AND q.ticker = u.ticker;
```

IF I comment out just one line from the conditions, the query works but it's obviously not what I want.

But the below query works just fine: 

```sql
WITH q AS (
  SELECT ticker, timestamp 
  FROM aggs 
  LATEST ON timestamp 
  PARTITION BY ticker
)
SELECT urls.ticker, q.timestamp, end, url FROM urls
LEFT JOIN q ON q.ticker = urls.ticker
WHERE q.timestamp <= urls.end;
```

So the issue only shows up on updates, not selects.
